From 8a6fd3a08642c2a5aac280554fc88d623d55b151 Mon Sep 17 00:00:00 2001
From: Nick Xie <nick@khadas.com>
Date: Wed, 1 Apr 2020 18:49:41 +0800
Subject: [PATCH 12/15] additional run_env_save sub for common board init

run_env_save is last before run_main_loop
this sub just check env_need_save and run saveenv if needed

Signed-off-by: Nick Xie <nick@khadas.com>
---
 common/board_r.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/common/board_r.c b/common/board_r.c
index 0bbeaa7594..8d786560ab 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -666,6 +666,16 @@ static int run_main_loop(void)
 	return 0;
 }
 
+static int run_env_save(void)
+{
+	if ( env_get("env_need_save") ){
+		printf("[i] env_need_save... ");
+		env_set("env_need_save",NULL);
+		run_command("saveenv",1);
+	}
+	return 0;
+}
+
 /*
  * We hope to remove most of the driver-related init and do it if/when
  * the driver is later used.
@@ -868,6 +878,7 @@ static init_fnc_t init_sequence_r[] = {
 #if defined(CONFIG_M68K) && defined(CONFIG_BLOCK_CACHE)
 	blkcache_init,
 #endif
+	run_env_save,
 	run_main_loop,
 };
 
-- 
2.17.1

